src/index.js

```diff
@@ -203,6 +211,24 @@ class HCaptcha extends React.Component {
       hcaptcha.execute(captchaId)
     }
 
+    async executeAsync () {
+      // Execute hCaptcha and wait for the callback
+      hcaptcha.execute();
+      
+      await this.promise;
```

Comment:
This will only work one time, you are going to want to generate the promise in here and store it. 

What you probably want to do is store the promise resolve here and return the promise from this function. Then in `handleSubmit` call to resolve with  the `{ token, ekey }`.

Reply:
Great catch! Will implement the changes soon